daffodil-test/src/test/resources/org/apache/daffodil/section12/lengthKind/PrefixedTests.tdml

```diff
@@ -0,0 +1,1820 @@
+  xmlns:xs="http://www.w3.org/2001/XMLSchema" 
+  xmlns:fn="http://www.w3.org/2005/xpath-functions"
+  xmlns:ex="http://example.com"
+  xmlns:tns="http://example.com"
+  xmlns:daf="urn:ogf:dfdl:2013:imp:daffodil.apache.org:2018:ext"
+  defaultRoundTrip="onePass">
+
+  <tdml:defineSchema name="lengthKindPrefixed-text.dfdl.xsd">
+
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+
+    <dfdl:format ref="ex:GeneralFormat"
+      representation="text"
+      encoding="UTF-8"
+      binaryDecimalVirtualPoint="2"
+      nilKind="literalValue"
+      nilValue="%WSP+;"
+      textBooleanTrueRep="TRUE"
+      textBooleanFalseRep="FALSE"
+      prefixIncludesPrefixLength="no" />
+
+    <xs:simpleType name="threeByteTextInt" dfdl:representation="text" dfdl:lengthKind="explicit" dfdl:length="3" dfdl:lengthUnits="bytes">
+      <xs:restriction base="xs:integer" /> 
+    </xs:simpleType>
+
+    <xs:simpleType name="threeByteBinInt" dfdl:representation="binary" dfdl:lengthKind="explicit" dfdl:length="3" dfdl:lengthUnits="bytes">
+      <xs:restriction base="xs:integer" />
+    </xs:simpleType>
+
+    <xs:simpleType name="twentyFourBitsTextInt" dfdl:representation="text" dfdl:lengthKind="explicit" dfdl:length="24" dfdl:lengthUnits="bits">
+      <xs:restriction base="xs:integer" /> 
+    </xs:simpleType>
+
+    <xs:simpleType name="twentyFourBitsBinInt" dfdl:representation="binary" dfdl:lengthKind="explicit" dfdl:length="24" dfdl:lengthUnits="bits">
+      <xs:restriction base="xs:integer" />
+    </xs:simpleType>
+
+    <xs:simpleType name="threeCharTextInt" dfdl:representation="text" dfdl:lengthKind="explicit" dfdl:length="3" dfdl:lengthUnits="characters" dfdl:textNumberJustification="left" dfdl:textPadKind="padChar" dfdl:textTrimKind="padChar" dfdl:textNumberPadCharacter="X">
+      <xs:restriction base="xs:integer" /> 
+    </xs:simpleType>
+
+    <xs:simpleType name="nByteTextInt" dfdl:representation="text" dfdl:lengthKind="prefixed" dfdl:prefixLengthType="ex:threeByteTextInt" dfdl:lengthUnits="bytes">
+      <xs:restriction base="xs:integer" />
+    </xs:simpleType>
+
+    <xs:element name="pl_text_string_txt_bytes" type="xs:string" dfdl:lengthKind="prefixed" dfdl:prefixLengthType="ex:threeByteTextInt" dfdl:lengthUnits="bytes" nillable="true" />
+    <xs:element name="pl_text_string_txt_bits" type="xs:string" dfdl:lengthKind="prefixed" dfdl:prefixLengthType="ex:twentyFourBitsTextInt" dfdl:lengthUnits="bits" nillable="true" />
+    <xs:element daf:parseUnparsePolicy="parseOnly" name="pl_text_string_txt_chars" type="xs:string" dfdl:lengthKind="prefixed" dfdl:prefixLengthType="ex:threeCharTextInt" dfdl:lengthUnits="characters" nillable="true" />
+
+    <xs:element name="pl_text_string_txt_bytes_includes" type="xs:string" dfdl:lengthKind="prefixed" dfdl:prefixLengthType="ex:threeByteTextInt" dfdl:lengthUnits="bytes" dfdl:prefixIncludesPrefixLength="yes" nillable="true" />
```

Comment:
Because of the line length here, I can't really code review this easily using the github tools. I have no horizontal scroll capability unless I scroll all the way to the bottom of the file.

Reply:
I can reformat to limit the line length.

Reply:
Done with a new fixup commit. Added some new defineFormat's that could be removed, reducing a lot of duplication. Wrapped lines to about 100, which seems to be about the magic number of characters that github will show on a review.